test/Kestrel.Core.Tests/HPackDecoderTests.cs

```diff
@@ -442,10 +443,28 @@ public void DecodesStringLength_GreaterThanLimit_Error()
                 .ToArray();
 
             var exception = Assert.Throws<HPackDecodingException>(() => _decoder.Decode(new ReadOnlySequence<byte>(encoded), endHeaders: true, handler: this));
-            Assert.Equal(CoreStrings.FormatHPackStringLengthTooLarge(4097, HPackDecoder.MaxStringOctets), exception.Message);
+            Assert.Equal(CoreStrings.FormatHPackStringLengthTooLarge(4097, MaxHeaderFieldSize), exception.Message);
             Assert.Empty(_decodedHeaders);
         }
 
+        [Fact]
+        public void DecodesStringLength_LimitConfigurable()
+        {
+            var decoder = new HPackDecoder(DynamicTableInitialMaxSize, 4097);
+            var string4097 = new string('a', 4097);
+
+            var encoded = _literalHeaderFieldWithoutIndexingNewName
+                .Concat(new byte[] { 0x7f, 0x82, 0x1f }) // 4097 encoded with 7-bit prefix, no Huffman encoding
```

Comment:
The `DecodesStringLength_GreaterThanLimit_Error` test is different.

```
var encoded = _literalHeaderFieldWithoutIndexingNewName
    .Concat(new byte[] { 0xff, 0x82, 0x1f }) // 4097 encoded with 7-bit prefix
```

Shouldn't these match? I'm guessing the old test is wrong since this test actually verifies the dictionary.